input-explainer.md

```diff
@@ -177,22 +177,34 @@ Most applications will want to visually represent the input sources somehow. The
   * `'tracked-pointer'`: If the `gripSpace` is not `null` an application-appropriate controller model should be drawn using that transform. If appropriate for the experience, the a visualization of the target ray and a cursor as described in the `'gaze'` should also be drawn.
   * `'screen'`: In all cases the point of origin of the target ray is obvious and no visualization is needed.
 
+Determining what to render is important as well. In some cases it's appropriate for the application to render a contextually appropriate model (such as a racket in a tennis game), in which case the physical appearance of the input source doesn't matter much. Many other times, though, it's desirable for the application to display a device that matches what the user is holding, especially when relaying instructions about it's use. Finally there are times when it's best to not render anything at all, such as when the XR device uses a transparent display and the user can see their hands and/or any tracked devices without app assistance.
+
+Controller devices should generally only be rendered if the `XRSession`'s `environmentBlendMode` is `'opaque'`, as any other mode implies that the user can see any physical device they may be holding. Rendering cursors or target rays may still be desirable.
+
+If the input source is a handheld device, the `XRInputSource` object should have a non-null `gamepad` attribute. The `Gamepad`'s `id` is used to determine what should be rendered if the app intends to visualize the input source itself, rather than an alternative virtual object. (See the section on [Button and Axis State](#button-and-axis-state) for more details.)
```

Comment:
Also do we want to state that there must be a non-null `gripSpace` if the gamepad isn't null?

Reply:
Hm............. What about a GearVR-style device where there's a touchpad on the device itself? Seems natural to want to expose that state via the `gamepad` attribute. Especially because that's already what happens with WebVR. (Probably wouldn't use the `xr-standard` mapping, though.)